windows/devices/uwb/UwbDeviceConnector.cxx

```diff
@@ -502,6 +502,8 @@ UwbConnector::HandleNotifications(std::stop_token stopToken)
                         hr = HRESULT_FROM_WIN32(lastError);
                         if (lastError == ERROR_INSUFFICIENT_BUFFER || lastError == ERROR_MORE_DATA) {
                             // Driver indicated buffer was too small and put required size in 'bytesRequired'. Retry with new size.
+                            const UWB_NOTIFICATION_DATA& notificationDataPartial = *reinterpret_cast<UWB_NOTIFICATION_DATA*>(std::data(uwbNotificationDataBuffer));

```

Comment:
If you wanted to avoid this cast, you could simply get the first 4 bytes of the buffer since that is where the size field is. Your way is probably safer, but this way would allow for some generalization (not in the notifications case, but for all of the other DDI structs). Since all of the DDI structs have 4-byte size fields in the beginning, this would allow us to avoid casting the buffer to the specific struct.

Reply:
I'm not willing to make any assumptions about a cross-process data structure, there's just too much risk. I have some other thoughts about the notification IOCTL that would affect this that I'll be bringing up this week. So, I'm going to wait and see how things shake out before generalizing the i/o with thread pools.

Reply:
I came up with a few ideas for how to achieve generalized i/o with this method. Will share offline.